src/components/PostsList.tsx

```diff
@@ -1,56 +1,59 @@
 import { Post } from '../types/Post';
+import { set as setSelectedPost } from '../features/selectedPost';
 
-type Props = {
-  posts: Post[],
-  selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
-};
-
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList = () => {
+  const { items } = useAppSelector(state => state.posts);
+  const dispatch = useAppDispatch();
+  const selectedPost = useAppSelector(
+    state => state.selectedPost,
+  ) as Post | null;
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-      <tbody>
-        {posts.map(post => (
-          <tr key={post.id} data-cy="Post">
-            <td data-cy="PostId">{post.id}</td>
-            <td data-cy="PostTitle">{post.title}</td>
-            <td className="has-text-right is-vcentered">
-              <button
-                type="button"
-                data-cy="PostButton"
-                className={classNames(
-                  'button',
-                  'is-link',
-                  {
-                    'is-light': post.id !== selectedPostId,
-                  },
-                )}
-                onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
-                }}
-              >
-                {post.id === selectedPostId ? 'Close' : 'Open'}
-              </button>
-            </td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
           </tr>
-        ))}
-      </tbody>
-    </table>
-  </div>
-);
+        </thead>
+
+        <tbody>
+          {items.map(post => (
+            <tr key={post.id} data-cy="Post">
+              <td data-cy="PostId">{post.id}</td>
+              <td data-cy="PostTitle">{post.title}</td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  type="button"
+                  data-cy="PostButton"
+                  className={classNames(
+                    'button',
+                    'is-link',
+                    {
+                      'is-light': post.id !== selectedPost?.id,
+                    },
+                  )}
+                  onClick={() => {
+                    if (selectedPost && selectedPost.id === post.id) {
+                      dispatch(setSelectedPost(null));
+                    } else {
+                      dispatch(setSelectedPost(post));
+                    }
+                  }}
```

Comment:
Move into separate handler instead of anonymous function in jsx
```
const onPostClick = (post) => () => {
  if (selectedPost && selectedPost.id === post.id) {
    dispatch(setSelectedPost(null));
  } else {
    dispatch(setSelectedPost(post));
  }
}

```
```suggestion
                  onClick={onPostClick(post)}
```